File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> answer = new ArrayList<Integer>();
        int i=1;
        for(;i<1024;i*=2){
            if(arr.length > i) continue;
            else if(arr.length == i) return arr;
            else break;
        }
        for(int j=0;j<arr.length;j++)
            answer.add(arr[j]);
        for(int j=arr.length;j<i;j++)
            answer.add(0);
        return answer.stream().mapToInt(k->k).toArray();
    }
}

File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    int i=1;
    for(;i<1024;i*=2){
        if(arr.size() > i) continue;
        else if(arr.size() == i) return arr;
        else break;
    }
    for(int j=arr.size();j<i;j++)
        arr.push_back(0);
    return arr;
}